gcp-logging/src/main/java/io/micronaut/gcp/logging/GoogleCloudPropertyDefiner.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.logging;
+
+import ch.qos.logback.core.PropertyDefinerBase;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Logback extension that sets a property called google_cloud_logging to allow users to switch between appender configurations.
+ *
+ * @author Vinicius Carvalho
+ * @since 3.1.1
+ */
+public class GoogleCloudPropertyDefiner extends PropertyDefinerBase {
+
+    private static final int DEFAULT_READ_TIMEOUT = 500;
+    private static final int DEFAULT_CONNECT_TIMEOUT = 500;
+    private static final String GOOGLE_COMPUTE_METADATA = "http://metadata.google.internal";
+
+    @Override
+    /**
+     * @return true if application is running on GCP
+     */
+    public String getPropertyValue() {
+        return isGoogleCompute().toString();
+    }
+
+    private Boolean isGoogleCompute() {
```

Comment:
Since this does a http call, does it affect a startup time?

Reply:
It will, the code is taken from the same way micronaut deduces a GCP environment, but good point there is a way to bypass this, (although I actually never set `micronaut.cloud.platform=gcp`) that should avoid the request, easy patch. Thanks for pointing

Reply:
Also worth to note that this won't get executed unless the user chooses to include the evaluation libs of janino and commons compiler. It does not get evaluated just for being present.
Because the logging context is outside the micronaut context, I did not found a way to reuse the environment detection from the framework itself. I may be wrong here, @graemerocher any way to tap into the Enviroment detection here?. Adding a bypass flag will certainly help here.

Reply:
Not from log config no. I am wondering abut janino and commons compiler. These approaches are unlikely to ever work on native image. Would it make sense to instead set `logback.configurationFile` to a GCP specific one for production?

See https://stackoverflow.com/questions/21885787/setting-logback-xml-path-programmatically

Reply:
Hum, looks like we will have to find a intermediate solution here. Seems that changing logback programatically would require it to take place before micronaut starts. 

I forgot about native-image, so I guess we should scratch that.

I know spring has an extension that they enable profiles via `<springProfile>` which then ties the configuration to the active profile.

I haven't tried but the property definer should be able to return the name of the appender instead, and that could return the JSON_APPENDER or an STDOUT and the included file defines both. This is just in case the user does not want to either explicit set or have some sort of pipeline that during application build include to correct log configuration file. I just thought that having some sort of automatic switch to be a nice touch.

Reply:
Seems like a good compromise to me